go/tools/builders/generate_checker_main.go

```diff
@@ -31,15 +31,206 @@ import (
 var codeTpl = `
 package main
 
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"os"
+	"path/filepath"
+	"strings"
+
+{{range $i, $importPath := .}}
+	_ "{{$importPath}}"
+{{- end}}
+
+	"github.com/bazelbuild/rules_go/analysis"
+	"golang.org/x/tools/go/gcexportdata"
+)
+
+type multiFlag []string
+
+func (m *multiFlag) String() string {
+	if len(*m) == 0 {
+		return ""
+	}
+	return fmt.Sprint(*m)
+}
+
+func (m *multiFlag) Set(v string) error {
+	(*m) = append(*m, v)
+	return nil
+}
+
+var (
+	archiveFiles multiFlag
+	stdlib       string
+)
+
+func init() {
+	flag.Var(&archiveFiles, "archivefile", "Archive file of a direct dependency")
+	flag.StringVar(&stdlib, "stdlib", "", "Root directory of stdlib")
+}
+
 func main() {
-	// This is a stub.
+	flag.Parse()
+	if stdlib == "" {
+		fmt.Fprintln(os.Stderr, "checker: missing stdlib root directory")
```

Comment:
Nit: Use `log.Printf` instead of using `fmt.Fprintf(os.Stderr, ...)` directly. It will automatically have a prefix, so you won't need to write `checker: ` here.